src/Resources/app/administration/src/module/sw-extension-store/service/extension-store-channel.service.ts

```diff
@@ -0,0 +1,146 @@
+import { handle } from '@shopware-ag/meteor-admin-sdk/es/channel';
+import ExtensionStoreActionService from 'src/module/sw-extension/service/extension-store-action.service';
+import ShopwareExtensionService from 'src/module/sw-extension/service/shopware-extension.service';
+import ExtensionStoreLicensesService from './extension-store-licenses.service'
+import type { Router } from 'vue-router';
+import { ShopwareMessageTypes } from '@shopware-ag/meteor-admin-sdk/es/message-types';
+
+type StoreChannelActionData = {
+  action: string;
+}
```

Comment:
This could be made a bit safer, unless we explicitly want it to be rather lax:

```suggestion
type StoreChannelAction = 'handshake' | 'routeTo' | 'purchase';

type StoreChannelActionData = {
  action: StoreChannelAction;
}
```